rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -0,0 +1,217 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.*;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
+import co.rsk.bitcoinj.store.BlockStoreException;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import co.rsk.peg.btcLockSender.BtcLockSenderProvider;
+import co.rsk.peg.constants.BridgeConstants;
+import co.rsk.peg.constants.BridgeMainNetConstants;
+import co.rsk.peg.federation.*;
+import co.rsk.peg.feeperkb.FeePerKbSupport;
+import co.rsk.peg.storage.BridgeStorageAccessorImpl;
+import co.rsk.peg.storage.StorageAccessor;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import co.rsk.test.builders.FederationSupportBuilder;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.core.*;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.*;
+
+import static org.mockito.Mockito.*;
+
+public class RegisterBtcTransactionIT {
+
+    private static final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+    private static final NetworkParameters btcParams = bridgeConstants.getBtcParams();
+    private final BridgeSupportBuilder bridgeSupportBuilder = BridgeSupportBuilder.builder();
+
+    @Test
+    void whenRegisterALegacyBtcTransactionTheBridgeShouldRegisterTheNewUtxoAndTransferTheRbtcBalance() throws BlockStoreException, AddressFormatException, IOException, BridgeIllegalArgumentException {
```

Comment:
The pattern we tend to follow is `methodName_when..._should...`. Since this is an integration test is fine to leave as blank the methodName part.

Note also that there is really not a need to declare all the types of exceptions since this is a test. If a test expects an exception to be thrown it should be asserted. So a simple `Exception` declaration will suffice. 

```suggestion
    void whenRegisteringALegacyBtcTransaction_shouldRegisterTheNewUtxoAndTransferTheRbtcBalance() throws Exception {
```

Reply:
Surrounding the methods with try/catch and fail the test in case of an exception, is it a valid approach?

Reply:
What I mean is that if the test expects an exception thrown, then we can assert it as `assertThrows(Exception.class, () -> methodThatThrowsException());`. But here since it is not expected we can declare it in the method signature `throws Exception`. This reduces boilerplate code since we know this test will not throw an exception since it is not intended.